homeassistant/components/insteon/api/logging.py

```diff
@@ -0,0 +1,47 @@
+"""Turn Insteon logging on and off."""
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import HomeAssistant
+
+from ..const import ID, TYPE
+
+MSG_LOGGER = logging.getLogger("pyinsteon.messages")
+TOPIC_LOGGER = logging.getLogger("pyinsteon.topics")
```

Comment:
Home Assistant already supports these features natively... why add-ons a separate API and handling for that?

Reply:
When you say HA already supports this, I am not sure how it does that other than to put it into `configuration.yaml` and restart. Is there a different method to turn a log on/off without restart?

Reply:
There are multiple approaches. There is a service available for setting logging on the fly.

But more importantly, if the loggers are defined in the `manifest.json` file of the integration, they can just be enabled/disabled on the integration config entry:

<https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics>